### DISCUSSION TITLE
mogrify -resize produces 3 copy of the image, and 1 is 3 pixel shorter o.O

### DISCUSSION THREAD

**Discussion by RoyBellingan** [ImageMagick/ImageMagick#6508] [Help] [answered]
### ImageMagick version

7.1.1-12

### Operating system, version and so on

Opensuse Tumbleweed (update 3 days ago)

### Description

For some reason the following image (not edited caming from a Nikon Z6II) 

https://drive.google.com/file/d/1s1JKR6wLh42tu2DC0TLuQUfL2dmeKaTB/view?usp=drive_link

If processed with the command 

mogrify -resize 1200x1200 -format jpeg original.JPG

Produces... 3 files ?

original-0.jpeg
original-1.jpeg

Have the same sha1sum

and the original-2.jpeg is... different is 3 pixel, smaller (1200x795 instead of 1200x798) compared to the other 2 and sensibly smaller in dimension (410K insted of 789K)

What is going on o.O

Same problem if using 

convert original.JPG -resize 1200x1200 -format jpeg resized.jpg

Same sha1sum is produced, and 3 images are outputted resized-0.jpg / resized-1.jpg / resized-2.jpg

I will try to download and compile on my machine to check this behaviour that is quite bizzare...

The whole folder 
https://drive.google.com/drive/folders/1Azo34Pms06AoqbteNoLXrbOzBhKQ3aC5?usp=sharing

*********
Update if instead I do this

convert original.JPG -format avif resized.avif

only one image is created, and is correct

convert original.avif -resize 200x200 -format avif resized2.avif

Again only one, really no idea why jpg are doomed in my machine ?

Also if I tried 
-format jpg (instead of -format jpeg) the triplication error is present

**Comment by fmw42** [comment]
I get the same result from 

`magick original.JPG -resize 1200x1200 x.jpg`

Perhaps this has to do with changes to read MPO JPG

**Reply by RoyBellingan** [reply]
Same error is happening also with a Niikon Z6 (for which I processed in the past 1000's of image)

I just compiled the code, and starting to debug.

**Comment by snibgo** [comment, marked as answer]
```
magick original.jpg info:

original.jpg[0] JPEG 6048x4024 6048x4024+0+0 8-bit sRGB 12912521B 0.094u 0:00.109
original.jpg[1] JPEG 6048x4024 6048x4024+0+0 8-bit sRGB 11605504B 0.094u 0:00.111
original.jpg[2] JPEG 640x424 640x424+0+0 8-bit sRGB 11605504B 0.094u 0:00.112
```

The file contains three images. (It is an MPF aka MPO file, containing a base image and two thumbnails.) If you want only the first image, use `[0]`:
```
mogrify -resize 1200x1200 -format jpeg original.JPG[0]
```

**Reply by RoyBellingan** [reply]
thank you very much! literally never saw this

**Comment by snibgo** [comment]
IM has only recently been enhanced to read all the MPF images in a jpeg file. It doesn't currently have the ability to *write* multiple images to jpeg files, so it creates one jpeg file per image.

**Reply by RoyBellingan** [reply]
Thank I will just leave as note that *I THINK* is the [1] file the correct one I have to read for this photocamera (at least is the one correctly rotated).

**Comment by dlemstra** [comment]
We have changed this behavior in version `7.1.1-12`. You should get a single image in that version? There was a short time window where you would get three images. You should now only get that when you rename `original.JPG` to `original.mpo`.

**Reply by RoyBellingan** [reply]
7.1.1-12 is the version I used. I have not yet tried more recent version, tried to compile in local but had error about the jpg support missing and more and.. just founded the solution meanwhile.

**Reply by RoyBellingan** [reply]
yes on version 7.1.1-15  the "problem" is gone, and I have to use format .mpo to have access to the 3 subfile
